src/ivc/protogalaxy/mod.rs

```diff
@@ -694,6 +698,34 @@ mod verify_chip {
             })
     }
 
+    pub fn verify_sps<C: CurveAffine, const L: usize>(
+        region: &mut RegionCtx<C::Base>,
+        ro_circuit: &mut impl ROCircuitTrait<C::Base>,
+        incoming: &[AssignedPlonkInstance<C>; L],
+    ) -> Result<(), Halo2PlonkError>
+    where
+        C::Base: FromUniformBytes<64> + PrimeFieldBits,
+        C::ScalarExt: FromUniformBytes<64> + PrimeFieldBits,
+    {
+        for pi in incoming {
+            if pi.challenges.is_empty() {
```

Comment:
double check: if one of them is empty, then all of them should be empty?

Reply:
In theory, yes, but since I don't want it to be implicit (it's not fixed at the type level in any way), I didn't put it out of cycle there.